backend/elasticsearch-client/jdk-rest-client/src/main/java/org/hibernate/search/backend/elasticsearch/client/jdk/cfg/spi/RestJdkClient.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.backend.elasticsearch.client.jdk.cfg.spi;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import org.hibernate.search.util.common.annotation.Incubating;
+import org.hibernate.search.util.common.impl.Closer;
+
+@Incubating
+public class RestJdkClient implements AutoCloseable {
```

Comment:
I put the client and node provider into the SPI package only because we usually say that "you can pass your own instance of the rest client to us" and if the client is in the impl package.. then it somewhat defeats that point